### PR TITLE
added nba_scraper function and documentation. Also edited Readme for Docker preparation.

### PR DESCRIPTION
**Pull request by kfoofw**
completed nba_scraper function with documentation and test scripts

### CODE REVIEW

File: R/nba_scraper.R [hunk 1]
```diff
@@ -1,26 +1,154 @@
...
-#' Scrape the tabular data from ESPN NBA website into a csv file.
+#' Scrape the tabular data from ESPN NBA website into a csv file using RSelenium.
 #'
-#' @param season_year string
+#' @param season_year int from 2001 to 2019 (upper limit based on latest year)
 #' @param season_type string
 #' @param csv_path string
+#' @param port int with L suffix
+#' @param sel_browser string
 #' @param nba_data_env_name string
```
Comment (AndresPitta) [inline R/nba_scraper.R]: The function should return the dataframe, this parameter should be changed

File: R/nba_scraper.R [hunk 2]
```diff
@@ -1,26 +1,154 @@
...
+#' \dontrun{
+#' #' nba_scraper(2018, season_type = "regular", "nba_2018.csv",
+#'             port=4445L, sel_browser = "chrome",
+#'             nba_data_env_name = "nba_2018_regular")
+#' }
+#'
+#' # Scrape playoffs season 2017/18 using "firefox" driver without
+#' # creating tibble in Global Env
+#' \dontrun{
+#' nba_scraper(2017, season_type = "playoffs", "nba_2017_playoffs.csv",
```
Comment (AndresPitta) [inline R/nba_scraper.R]: I think the name must be alligned to what we see in the webpage
```suggestion
#' nba_scraper(2017, season_type = "postseason", "nba_2017_postseason.csv",
```

File: R/nba_scraper.R [hunk 3]
```diff
@@ -1,26 +1,154 @@
...
+#' }
+nba_scraper <- function(season_year = 2018, season_type = "regular", csv_path = "nba_2018.csv", port=4445L, sel_browser = "chrome", nba_data_env_name = NULL) {
+
+  # Check season_year is not integer or not within range
+  if ((round(season_year) != season_year) & (season_year < 2001) & (season_year > 2019)){
+    stop("'season_year' must be an integer between 2001 to 2019")
+  }
+
+  # Check season_type is not "regular" or "playoffs"
+  if ((season_type != "playoffs") & (season_type != "regular")){
```
Comment (AndresPitta) [inline R/nba_scraper.R]:
```suggestion
  if ((season_type != "postseason") & (season_type != "regular")){
```

File: R/nba_scraper.R [hunk 4]
```diff
@@ -1,26 +1,154 @@
...
+
+  # Set up parameters for remoteDriver
+  remDr <- RSelenium::remoteDriver(
+    remoteServerAddr = "localhost",
+    port = port,
+    browserName = sel_browser
+  )
+
+  # Print message
+  print("Scraping commencing. Please wait!")
```
Comment (AndresPitta) [inline R/nba_scraper.R:85]: Good addition!

File: R/nba_scraper.R [hunk 5]
```diff
@@ -1,26 +1,154 @@
...
+    compiled_df[i,2] <- player_team
+    compiled_df[i,3:23] <- stats_list
+  }
+  # Write to csv
+  write.csv(compiled_df, csv_path)
+
+  # If nba_data_env_name is given, create a tibble in Global Env
+  if (!is.null(nba_data_env_name)) {
+    assign(nba_data_env_name, tibble::tibble(compiled_df), envir = .GlobalEnv)
+  }
```
Comment (AndresPitta) [inline R/nba_scraper.R]: Instead of assinging the object to the environment we should return the dataframe

File: README.md [hunk 6]
```diff
@@ -42,10 +42,81 @@ Functions
...
+port, we can proceed with testing
+
+    library(rsketball)
+    # Scrape regular season 2018/19 using "chrome" driver and
+    # create "nba_2018_regular" tibble in Global Env
+    nba_scraper(season_year = 2018, season_type = "regular", 
+                csv_path = "nba_2018_regular.csv",
+                port=4445L, # Insert the port number with a L suffix as shown
+                sel_browser = "chrome",
+                nba_data_env_name = "nba_2018_regular")
```
Comment (AndresPitta) [inline README.md]: Remember removing the environment parameter from the documentation

File: README.md [hunk 7]
```diff
@@ -22,8 +22,8 @@ Functions
 `nba_scraper`
 
 -   Scrapes data from ESPN NBA data into a csv file. User can specify
-    the year of the season (2016, 2017, etc) and the season type
-    (regular or playoffs).
+    the year of the season (2001 to 2019 at the moment) and the season
+    type (“regular” or “playoffs”).
```
Comment (AndresPitta) [inline README.md]:
```suggestion
    type (“regular” or “postseason”).
```

File: README.Rmd [hunk 8]
```diff
@@ -29,7 +29,7 @@ https://www.espn.com/nba/stats/player/_/season/2019/seasontype/2
 `nba_scraper`
   
   * Scrapes data from ESPN NBA data into a csv file. User can specify the year of the season
-  (2016, 2017, etc) and the season type (regular or playoffs).
+  (2001 to 2019 at the moment) and the season type ("regular" or "playoffs").
```
Comment (AndresPitta) [inline README.Rmd]:
```suggestion
  (2001 to 2019 at the moment) and the season type ("regular" or "postseason").
```

File: README.Rmd [hunk 9]
```diff
@@ -46,9 +46,60 @@ https://www.espn.com/nba/stats/player/_/season/2019/seasontype/2
...
+
+```
+library(rsketball)
+# Scrape regular season 2018/19 using "chrome" driver and
+# create "nba_2018_regular" tibble in Global Env
+nba_scraper(season_year = 2018, season_type = "regular", 
+            csv_path = "nba_2018_regular.csv",
+            port=4445L, # Insert the port number with a L suffix as shown
+            sel_browser = "chrome",
+            nba_data_env_name = "nba_2018_regular")
```
Comment (AndresPitta) [inline README.Rmd]: Remember removing the environtment variable

**Review by AndresPitta — CHANGES_REQUESTED** [review]

File: R/nba_scraper.R (hunk 5 above)
Comment (kfoofw) [reply R/nba_scraper.R]: this has been changed accordingly with a more recent commit!

File: R/nba_scraper.R (hunk 3 above)
Comment (kfoofw) [reply R/nba_scraper.R]: this has been changed accordingly with a more recent commit!

File: R/nba_scraper.R (hunk 2 above)
Comment (kfoofw) [reply R/nba_scraper.R]: this has been changed accordingly with a more recent commit!

File: R/nba_scraper.R (hunk 1 above)
Comment (kfoofw) [reply R/nba_scraper.R]: this has been changed accordingly with a more recent commit!

File: README.md (hunk 6 above)
Comment (kfoofw) [reply README.md]: this has been changed accordingly with a more recent commit!

File: R/nba_scraper.R (hunk 4 above)
Comment (kfoofw) [reply R/nba_scraper.R:85]: thank you!

File: README.md (hunk 7 above)
Comment (kfoofw) [reply README.md]: this has been changed accordingly with a more recent commit!

File: README.Rmd (hunk 8 above)
Comment (kfoofw) [reply README.Rmd]: this has been changed accordingly with a more recent commit!

**Review by AndresPitta — APPROVED** [review]
Changes reviewed, everything is nice!

**Pull request merged**